src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/RemoteAppAuthenticationModule.Framework.cs

```diff
@@ -28,10 +28,36 @@ void MapRemoteAuthenticationHandler(object sender, EventArgs e)
             if (string.Equals(context.Request.Path, _options.AuthenticationEndpointPath, StringComparison.OrdinalIgnoreCase)
                 && context.Request.HttpMethod.Equals("GET", StringComparison.OrdinalIgnoreCase))
             {
-                if (!string.Equals(_options.RemoteServiceOptions.ApiKey, context.Request.Headers.Get(_options.RemoteServiceOptions.ApiKeyHeader), StringComparison.Ordinal))
+                var apiKey = context.Request.Headers.Get(_options.RemoteServiceOptions.ApiKeyHeader);
+                if (apiKey is null)
```

Comment:
Can you add some tests around this? This logic should be safeguarded behind unit tests so we make sure not to regress

Reply:
Yep, I agree we should have some tests. I'm out next week, but will get this updated with unit tests when I'm back after the 4th.

Reply:
Done